### PR TITLE
gotrue{,-supabase}: init

### LINKED ISSUE #170203: GoTrue
**Project description**
Hoping to package GoTrue: https://github.com/supabase/gotrue

Will do it if there's no one else who's keen to take it up

Relates to #159878

**Metadata**

* homepage URL: https://github.com/supabase/gotrue
* source URL: https://github.com/supabase/gotrue
* license: Apache2
* platforms: unix, linux, darwin

### PR DESCRIPTION
**Pull request by urandom2**
###### Description of changes

the explicit packaging request asked for [supabase/gotrue](https://github.com/supabase/gotrue), but this is a fork of an upstream tool. as such, we have added both, but chosen to name the requested package `gotrue-supabase`, as it is a fork.

Fixes #170203

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by gador** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 203676` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>gotrue</li>
<li>gotrue-supabase</li>
</ul>
</details>

File: pkgs/tools/security/gotrue/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{ buildGoModule, fetchFromGitHub, lib }:
```
Comment (figsoda) [inline pkgs/tools/security/gotrue/default.nix:1]:
```suggestion
{ lib, buildGoModule, fetchFromGitHub }:
```

File: pkgs/tools/security/gotrue/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    rev = "v${version}";
+    hash = "sha256-9h6CyCY7741tJR+qWDLwgPkAtE/kmaoTqlXEY+mOW58=";
+  };
+
+  vendorHash = "sha256-x96+l9EBzYplGRFHsfQazSjqZs35bdXQEJv3pBuaJVo=";
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X github.com/netlify/gotrue/cmd.Version=${version}"
```
Comment (figsoda) [inline pkgs/tools/security/gotrue/default.nix:19]:
```suggestion
    "-X=github.com/netlify/gotrue/cmd.Version=${version}"
```

File: pkgs/tools/security/gotrue/supabase.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
+{ buildGoModule, fetchFromGitHub, lib }:
```
Comment (figsoda) [inline pkgs/tools/security/gotrue/supabase.nix:1]:
```suggestion
{ lib, buildGoModule, fetchFromGitHub }:
```

File: pkgs/tools/security/gotrue/supabase.nix [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    rev = "v${version}";
+    hash = "sha256-uFE2pcEpGhrl8LcZLvYEMlq8sgRmFkltf3H8huZzXpM=";
+  };
+
+  vendorHash = "sha256-uchgHxUczb4IIUkUdHWyeXWr2LXda6eWwjQxUBcPDoA=";
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X github.com/netlify/gotrue/utilities.Version=${version}"
```
Comment (figsoda) [inline pkgs/tools/security/gotrue/supabase.nix:19]:
```suggestion
    "-X=github.com/netlify/gotrue/utilities.Version=${version}"
```

**Review by figsoda — APPROVED** [review]
lgtm, just some nitpicks

**Review by figsoda — APPROVED** [review]

**Pull request merged**